src/storage/src/storage_state.rs

```diff
@@ -834,20 +832,31 @@ impl<'w, A: Allocate> Worker<'w, A> {
     /// subscribe response buffer. We will need to be vigilant with future
     /// modifications to `StorageState` to line up changes there with clean
     /// resets here.
+    ///
+    /// Returns `true` if the client (through the `CommandReceiver`) has
+    /// been disconnected.
+    #[must_use]
     fn reconcile(
         &mut self,
         internal_cmd_tx: &mut dyn InternalCommandSender,
         async_worker: &mut AsyncStorageWorker<mz_repr::Timestamp>,
         command_rx: &CommandReceiver,
-    ) {
+    ) -> bool {
```

Comment:
I would make this return a `Result` and let the caller interpret it as a `disconnected` flag

Reply:
that definitely cleaner!

Reply:
updated with this change!